ring/batch.go

```diff
@@ -112,29 +129,45 @@ func DoBatch(ctx context.Context, op Operation, r ReadRing, keys []uint32, callb
 }
 
 func (b *batchTracker) record(sampleTrackers []*itemTracker, err error) {
-	// If we succeed, decrement each sample's pending count by one.  If we reach
-	// the required number of successful puts on this sample, then decrement the
-	// number of pending samples by one.  If we successfully push all samples to
-	// min success instances, wake up the waiting rpc so it can return early.
-	// Similarly, track the number of errors, and if it exceeds maxFailures
-	// shortcut the waiting rpc.
+	// If we reach the required number of successful puts on this sample, then decrement the
+	// number of pending samples by one.
```

Comment:
This comment uses the term `samples`, but isn't that term overly specific to the use case of `ring.DoBatch()` in the ingester? `ring.DoBatch()` could also be used for other things than sending samples, so it might make sense to just use the term `item` here instead of `sample` because that term would also match with the naming of the type `itemTracker`.